invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/files.html

```diff
@@ -90,11 +95,25 @@
 {%- endmacro %}
 
 
-{% macro file_list_box(files, pid, is_preview) %}
+{% macro file_list_box(files, pid, is_preview, record_access, format_date) %}
+  {% if record_access.files == 'restricted' and  record_access.embargo.active%}
```

Comment:
Three times refactor? The decision logic to determine if we are in a Public Public / Public Restricted / Public Embargoed / Restricted / Embargoed is repeated multiple times at this point. Perhaps an abstraction that returns the current "protection case" can be used to then have the conditionals on (or it could incorporates the branches...)

Reply:
Yes, very good idea, thanks!